core/pta/k3/otp.c

```diff
@@ -0,0 +1,114 @@
+// SPDX-License-Identifier: BSD-2-Clause
+/*
+ * Texas Instruments System Control Interface Driver
+ *
+ * Copyright (C) 2023 Texas Instruments Incorporated - https://www.ti.com/
+ *	Manorit Chawdhry <m-chawdhry@ti.com>
+ */
+
+#include <drivers/ti_sci.h>
+#include <inttypes.h>
+#include <k3/otp_keywriting_ta.h>
+#include <kernel/pseudo_ta.h>
+
+static TEE_Result write_otp_row(uint32_t param_types, TEE_Param params[4])
+{
+	TEE_Result ret = 0;
```

Comment:
`ti_sci_write_otp_row()` return an `int`. This function returns a `TEE_Result`. Something is weird here.
The issue comes from `ti_sci_xxx()` and friends that return `TEE_Result` compliant codes yet using `int` type return values. It should be cleaned one day or another.